src/main/java/org/broadinstitute/consent/http/service/DatabaseElectionAPI.java

```diff
@@ -568,46 +577,45 @@ private Document describeDataAccessRequestById(String id){
     private void setGeneralFields(Election election, String referenceId, ElectionType electionType) {
         election.setCreateDate(new Date());
         election.setReferenceId(referenceId);
-        BasicDBObject query;
-        Document dar;
+        election.setElectionType(electionType.getValue());
+
         switch (electionType) {
             case TRANSLATE_DUL:
-                election.setElectionType(ElectionType.TRANSLATE_DUL.getValue());
                 Consent consent = consentDAO.findConsentById(referenceId);
                 election.setTranslatedUseRestriction(consent.getTranslatedUseRestriction());
                 election.setUseRestriction(consent.getUseRestriction());
                 election.setDataUseLetter(consent.getDataUseLetter());
                 election.setDulName(consent.getDulName());
                 break;
             case DATA_ACCESS:
-                election.setElectionType(ElectionType.DATA_ACCESS.getValue());
-                query = new BasicDBObject(DarConstants.ID, new ObjectId(referenceId));
-                dar = mongo.getDataAccessRequestCollection().find(query).first();
-                election.setTranslatedUseRestriction(dar.getString(DarConstants.TRANSLATED_RESTRICTION));
-                try {
-                    String restriction  =  new Gson().toJson(dar.get(DarConstants.RESTRICTION, Map.class));
-                    election.setUseRestriction((UseRestriction.parse(restriction)));
-                } catch (IOException e) {
-                    election.setUseRestriction(null);
-                }
-                break;
             case RP:
-                election.setElectionType(ElectionType.RP.getValue());
-                query = new BasicDBObject(DarConstants.ID, new ObjectId(referenceId));
-                dar = mongo.getDataAccessRequestCollection().find(query).first();
+                BasicDBObject query = new BasicDBObject(DarConstants.ID, new ObjectId(referenceId));
+                Document dar = mongo.getDataAccessRequestCollection().find(query).first();
+                List<?> datasetIdList = dar.get(DarConstants.DATASET_ID, List.class);
+                if (datasetIdList != null && !datasetIdList.isEmpty()) {
```

Comment:
This is the meat of this PR. Find the `datasetId` field of the data access request and populate that value in the election. DARs are mongo docs, elections are relational db rows and we need to sync the two here. For historical reasons, the `datasetId` field is actually a list of values, but should only ever contain a single value.